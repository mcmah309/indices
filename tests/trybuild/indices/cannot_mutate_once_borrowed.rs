use indices::indices;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four,) = indices!(&mut slice, 4);
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}");
}

fn two() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four, one) = indices!(&mut slice, 4, 1);
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}, {one}");
}

fn three() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four, _one, _two) = indices!(&mut slice, 4, 1, 2);
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}");
}

fn four() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four, one, two, three) = indices!(&mut slice, 4, 1, 2, 3);
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}, {one}, {two}, {three}");
}

fn five() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four, one, two, three, zero) = indices!(&mut slice, 4, 1, 2, 3, 0);
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}, {one}, {two}, {three}, {zero}");
}
