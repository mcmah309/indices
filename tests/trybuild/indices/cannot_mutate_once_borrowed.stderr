error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:14:5
   |
13 |     let (four,) = indices!(&mut slice, 4);
   |                            ---------- `slice` is borrowed here
14 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
15 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:14:5
   |
13 |     let (four,) = indices!(&mut slice, 4);
   |                            ---------- `slice` is borrowed here
14 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
15 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:24:5
   |
23 |     let (four, one) = indices!(&mut slice, 4, 1);
   |                                ---------- `slice` is borrowed here
24 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
25 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:24:5
   |
23 |     let (four, one) = indices!(&mut slice, 4, 1);
   |                                ---------- `slice` is borrowed here
24 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
25 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:34:5
   |
33 |     let (four, _one, _two) = indices!(&mut slice, 4, 1, 2);
   |                                       ---------- `slice` is borrowed here
34 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
35 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:34:5
   |
33 |     let (four, _one, _two) = indices!(&mut slice, 4, 1, 2);
   |                                       ---------- `slice` is borrowed here
34 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
35 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:44:5
   |
43 |     let (four, one, two, three) = indices!(&mut slice, 4, 1, 2, 3);
   |                                            ---------- `slice` is borrowed here
44 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
45 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:44:5
   |
43 |     let (four, one, two, three) = indices!(&mut slice, 4, 1, 2, 3);
   |                                            ---------- `slice` is borrowed here
44 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
45 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:54:5
   |
53 |     let (four, one, two, three, zero) = indices!(&mut slice, 4, 1, 2, 3, 0);
   |                                                  ---------- `slice` is borrowed here
54 |     slice[4] *= 2;
   |     ^^^^^^^^ use of borrowed `slice`
55 |     *four *= 2;
   |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
  --> tests/trybuild/indices/cannot_mutate_once_borrowed.rs:54:5
   |
53 |     let (four, one, two, three, zero) = indices!(&mut slice, 4, 1, 2, 3, 0);
   |                                                  ---------- `slice` is borrowed here
54 |     slice[4] *= 2;
   |     ^^^^^^^^^^^^^ use of borrowed `slice`
55 |     *four *= 2;
   |     ---------- borrow later used here
